File: 11-container-with-most-water/container-with-most-water.js
/**
 * @param {number[]} height
 * @return {number}
 */
var maxArea = function(height) {
    let max = 0;
    let i = 0;
    let j = height.length - 1;
    while(i < j){
        let width = j-i;
        let h = Math.min(height[i] , height[j]);
        let area = h * width;
        max = Math.max(max , area);
        if(height[i] < height[j]){
            i++;
        }else{
            j--;
        }
    }
    return max;
};

File: 11-container-with-most-water/container-with-most-water.java
class Solution {
    public int maxArea(int[] height) {
        int i = 0, j = height.length -1, area = 0, maxArea = 0;
        
        while (i < j) {
            int h = Math.min(height[i], height[j]);
            area = h * Math.abs(i - j);
            maxArea = Math.max(area, maxArea);
            while (i<j && height[i] <= h) {
                i++;
            }
            while (i<j && height[j] <= h) {
                j--;
            }
        }
        return maxArea;
    }
}